cpp/order/HexaticTranslational.cc

```diff
@@ -37,7 +37,14 @@ void HexaticTranslational<T>::computeGeneral(Func func, const freud::locality::N
             }
             if (normalize_by_k)
             {
-                m_psi_array[i] /= std::complex<float>(m_k);
+                if (total_weight == 0.)
+                {
+                    m_psi_array[i] /= std::complex<float>(total_weight);
```

Comment:
Force this to be `nan` by dividing by zero. Is there a better way to do this? I saw `std::nanf` but I wasn't sure what to pick as the argument.

Reply:
I haven't reviewed this PR thoroughly yet, but I'm not sure this is generating the output you want. `std::complex<float>(1)/std::complex<float>(0)` will return (inf, nan), not just nan ([here's an example](https://godbolt.org/z/dr9K3nYa8))

Reply:
So tried to set `m_psy_array` equal to just nan like this:

```python
m_psi_array[i] = std::complex<float>(std::nanf)
```

and then like this

```python
m_psi_array[i] = std::complex<float>(std::nanf)
```

All tests pass on my laptop, but fail on CI, so I reverted back to a divide by zero. I double checked the values from the no neighbor test for the Translational order parameter, and in python, it registers as nan and not inf.